### ISSUE TITLE
cPickle serialization problem for WDL Task objects

### ISSUE THREAD

**Issue by scottfrazer** [BD2KGenomics/toil#556]
I get a stack trace when trying to pass a [Call object](https://github.com/broadinstitute/pywdl/blob/master/wdl/binding.py#L178) into a wrapped function in Toil:

```
Traceback (most recent call last):
  File "pickle_bug.py", line 46, in <module>
    print(Job.Runner.startToil(root, options))
  File "/Users/sfrazer/venv/wdl-cwl-toil/lib/python2.7/site-packages/toil/job.py", line 436, in startToil
    rootJob = job._serialiseFirstJob(jobStore)
  File "/Users/sfrazer/venv/wdl-cwl-toil/lib/python2.7/site-packages/toil/job.py", line 1248, in _serialiseFirstJob
    self._serialiseJobGraph(jobWrapper, jobStore, None, True)
  File "/Users/sfrazer/venv/wdl-cwl-toil/lib/python2.7/site-packages/toil/job.py", line 1216, in _serialiseJobGraph
    job._serialiseJob(jobStore, jobsToJobWrappers, jobWrapper)
  File "/Users/sfrazer/venv/wdl-cwl-toil/lib/python2.7/site-packages/toil/job.py", line 1178, in _serialiseJob
    cPickle.dump(self, fileHandle, cPickle.HIGHEST_PROTOCOL)
TypeError: 'NoneType' object is not callable
```

This is the code to reproduce: (requires `pip install wdl`)

``` python
from toil.job import Job
import wdl, subprocess, io, re

wdl_code = """
task greeter {
  String name

  command {
    echo "Hello, ${name}"
  }
  output {
    String greeting = read_string(stdout())
  }
}

workflow test {call greeter}
"""

wdl_inputs = {
    "test.greeter.name": "World"
}

wdl_document = wdl.loads(wdl_code)

if len(wdl_document.workflows) != 1:
    sys.stderr.write('Current implementation requires exactly one workflow definition per WDL file\n')
    sys.exit(-1)

workflow = wdl_document.workflows[0]

def wrap_task(job, wdl_task):pass
def rootJob(job): pass

toil_tasks = {}
for task in wdl_document.tasks:
    toil_tasks[task.name] = Job.wrapJobFn(wrap_task, task)

if __name__ == '__main__':
    options = Job.Runner.getDefaultOptions("./toilWorkflow")
    root = Job.wrapJobFn(rootJob)
    for call in workflow.calls():
    root.addChild(toil_tasks[call.name])
    print(Job.Runner.startToil(root, options))
```

I imagine there's something about `cPickle` that I don't understand that I need to change in PyWDL

**Comment by scottfrazer** [comment]
It seems as though I was able to get around this issue by adding this line as the last line to all places where I implement `__getattr__` in the PyWDL code:

```python
return self.__getitem__(name)
```

**Comment by hannes-ucsc** [comment]
Return `super(Scope, self).__getattr__(name)` would be more logical IMHO.

**Comment by scottfrazer** [comment]
Yeah, I'll try that.  I may have spoke too soon, because soon after I tried my solution I got some infinite loops

**Comment by hannes-ucsc** [comment]
@scottfrazer, I've unassigned this  until we hear from you about the prospective workaround/fix.

**Comment by DailyDreaming** [comment]
Addressed in #1823.
